azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -43,16 +45,20 @@
 /**
  * This servlet exposes the REST APIs such as create, get etc. for image type. Below are the
  * supported APIs. Create Image Type API: POST /imageTypes?session.id=? --data @payload.json Get
- * Image Type API: GET /imageTypes?session.id=?&imageType=? GET /imageTypes/{id}?session.id=?
+ * Image Type API: GET /imageTypes?&imageType=? GET /imageTypes/{imageType}
```

Comment:
Nit: Extra & in GET /imageTypes?&imageType=?

Reply:
I believe one is a path parameter and one is a filter parameter. Unless we don't want to do that?

Reply:
Is the name of an image type unique? 
If not then the URI `/imageTypes/{imageType}` can represent 2 or more resources at the same time which is ambiguous and incorrect.
If yes then supporting the URI `/imageTypes/{imageType}` means that the name cannot be changed probably not even to correct typos because that would change the URI and break any code using the old one. How are we planning to support renames in that case?
Why `GET /imageTypes?&imageType=?` isn't enough for this use case?

Reply:
Image type name should be unique. Updates should be possible for renaming with put/post. In regards to the URI, my understanding is that renaming for a filter parameter and a path parameter would break things equally. The users don't have access to IDs, so using an ID is unrealistic.